server/connectors/api-connector-jdbc/src/main/scala/com/prisma/api/connector/jdbc/database/FilterConditionBuilder.scala

```diff
@@ -101,15 +111,31 @@ trait FilterConditionBuilder extends BuilderBase {
           .select(relationColumn(relationField))
           .from(relationTable(relation))
           .where(condition.invert(invertConditionOfSubSelect))
-
+      case nested: ScalarFilter =>
+        if (nested.field.name.equals("id")) {
+          val condition = nested.condition match {
```

Comment:
we could check on other condition operator, but they don't specifically make sense with id